python/lsst/ip/isr/shutterMotion.py

```diff
@@ -0,0 +1,589 @@
+                    "actionDuration": self.metadata["actionDuration"],
+                    "side": self.metadata["side"],
+                    "isOpen": self.metadata["isOpen"],
+                    "encodeSamples": self.writeEncodeSamples(),
+                    "hallTransitions": self.writeHallTransitions(),
+                    "fitResults": self.writeFitResults(),
+                },
+            }
+        elif formatVersion == 2.0:
+            outDict = {
+                "fileName": self.metadata["fileName"],
+                "fileType": self.metadata["fileType"],
+                "metadata": {
+                    "CALIBCLS": "lsst.ip.isr.ShutterMotionProfile",
+                    "OBSTYPE": self._OBSTYPE,
+                },
+                "obsId": self.metadata["obsId"],
+                "version": self.metadata.get("version", -1),
+                "motionProfile": {
+                    "startTime": {
+                        "tai": {
+                            "isot": self.metadata["startTime_tai"],
+                            "mjd": self.metadata["startTime_mjd"],
+                        },
+                    },
+                    "startPosition": self.metadata["startPosition"],
+                    "targetPosition": self.metadata["targetPosition"],
+                    "endPosition": self.metadata["endPosition"],
+                    "targetDuration": self.metadata["targetDuration"],
+                    "actionDuration": self.metadata["actionDuration"],
+                    "side": self.metadata["side"],
+                    "isOpen": self.metadata["isOpen"],
+                    "encodeSamples": self.writeEncodeSamples(),
+                    "hallTransitions": self.writeHallTransitions(),
+                    "fitResults": self.writeFitResults(),
+                },
+            }
+        else:
+            raise RuntimeError(f"Unknown file version: {formatVersion}")
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration, after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.  For shutter motion profiles, the first table
+            contains the samples, the second the Hall transition data,
+            and the third the model fits.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.ShutterMotionProfile`
+            The calibration defined in the tables.
+        """
+        samples = tableList[0]
+        transitions = tableList[1]
+        modelFits = tableList[2]
+
+        metadata = samples.meta
+
+        calib = cls()
+        calib.time_tai = samples["TIME_TAI"]
```

Comment:
Do all of these need a `[0]` at the end? Maybe.  But I don't understand how the round trip test works, unless it's incomplete.